geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert/common/TransformersTest.scala

```diff
@@ -274,7 +274,160 @@ class TransformersTest extends Specification {
           exp.eval(Array("", "1.0", "1.0")) must beFalse
           exp.eval(Array("", "2.0", "1.0")) must beTrue
         }
+      }
 
+      "string2 functions" >> {
+        "string2double" >> {
+          "double string2double zero default" >> {
+            val exp: Predicate = Transformers.parsePred("doubleEq(string2double($1, $3), string2double($2, $3))")
+            exp.eval(Array("", "1.2", "1.0", 0.0)) must beFalse
+            exp.eval(Array("", "0.0", "0.0", 0.0)) must beTrue
+          }
+          "double empty string string2double zero default" >> {
+            val exp: Predicate = Transformers.parsePred("doubleEq(string2double($1, $3), string2double($2, $3))")
+            exp.eval(Array("", "", "1.0", 0.0)) must beFalse
+            exp.eval(Array("", "", "0.0", 0.0)) must beTrue
+          }
+          "double null string string2double zero default" >> {
+            val exp = Transformers.parsePred("doubleEq(string2double($1,$3), string2double($2,$3))")
+            exp.eval(Array("", null, "1.0", 0.0)) must beFalse
+            exp.eval(Array("", null, "0.0", 0.0)) must beTrue
+          }
+          "double string2double null default" >> {
+            val exp: Predicate = Transformers.parsePred("doubleEq(string2double($1, $3), string2double($2, $3))")
+            exp.eval(Array("", "1.2", "1.0", null)) must beFalse
+            exp.eval(Array("", "0.0", "0.0", null)) must beTrue
+          }
+          "double empty string string2double null default" >> {
+            val exp: Predicate = Transformers.parsePred("doubleEq(string2double($1, $3), string2double($2, $3))")
+            exp.eval(Array("", "", "1.0", null)) must beFalse
+            exp.eval(Array("", "", null, null)) must beTrue
+          }
+          "double null string string2double null default" >> {
+            val exp = Transformers.parsePred("doubleEq(string2double($1,$3), string2double($2,$3))")
+            exp.eval(Array("", null, "1.0", null)) must beFalse
+            exp.eval(Array("", null, null, null)) must beTrue
+          }
+        }
+        "string2int" >> {
+          "int string2int zero default" >> {
+            val exp: Predicate = Transformers.parsePred("intEq(string2int($1, $3), string2int($2, $3))")
```

Comment:
It might be better to just evaluate the string2int for a couple different inputs, and not use the intEq. That way you're checking the direct output. Would also require less tests.
e.g. for string to int, really you just need: "string2int($1, '0'::int)" for valid value, null string, empty string
Ditto other tests.